File: Data Structures & Algorithms/remove-element/submission-1.java
class Solution {
    public int removeElement(int[] nums, int val) {
        Queue<Integer> index = new LinkedList<>();
        int ans = 0;
        for(int i = 0; i<nums.length; i++)
        {
            if(nums[i]==val)
            {
                index.add(i);
                
            } else {
                ans++;
                if(index.isEmpty())
                {

                } else{
                    nums[index.poll()] = nums[i];
                    index.add(i);
                }

            }

        }
        return ans;
        
    }
}

File: Data Structures & Algorithms/products-of-array-discluding-self/submission-1.cpp
class Solution {
public:
    vector<int> productExceptSelf(vector<int>& nums) {
        int front = 1;
        int back = 1;
        int j = 0;
        vector<int> ans (nums.size());
        for(int i = 0; i < nums.size(); i++)
        {
            while(j<nums.size())
            {
                if(i>j)
                {
                    front*=nums[j];
                    
                } else if(i<j)
                {
                    back*=nums[j];
                }
                j++;
            }
            ans[i]=front*back;
            j=i+1;
            back = 1;
            front*=nums[i];
        }
        
        return ans;

    }
};
